jetty-io/src/main/java/org/eclipse/jetty/io/ArrayRetainableByteBufferPool.java

```diff
@@ -430,19 +433,23 @@ public void dump(Appendable out, String indent) throws IOException
 
     private Pool<RetainableByteBuffer>.Entry findOldestEntry(long now, Pool<RetainableByteBuffer> bucket)
     {
+        // This method may be in the hot path, do not use Java streams.
+
         RetainedBucket.Entry oldestEntry = null;
+        RetainableByteBuffer oldestBuffer = null;
+        long oldestAge = 0;
         for (RetainedBucket.Entry entry : bucket.values())
         {
-            if (oldestEntry != null)
-            {
-                long entryAge = NanoTime.elapsed(entry.getPooled().getLastUpdate(), now);
-                if (entryAge > NanoTime.elapsed(oldestEntry.getPooled().getLastUpdate(), now))
-                    oldestEntry = entry;
-            }
-            else
-            {
-                oldestEntry = entry;
-            }
+            RetainableByteBuffer buffer = entry.getPooled();
+            // Reserved but not acquired yet? Try next.
+            if (buffer == null)
+                continue;
+            long age = NanoTime.elapsed(buffer.getLastUpdate(), now);
+            if (oldestBuffer != null && age < oldestAge)
+                continue;
+            oldestEntry = entry;
+            oldestBuffer = buffer;
+            oldestAge = age;
```

Comment:
There you go... you can be both consistent and readable:
```suggestion
            if (buffer != null)
            {
                long age = NanoTime.elapsed(buffer.getLastUpdate(), now);
                if (oldestBuffer == null || age > oldestAge)
                {
                    oldestEntry = entry;
                    oldestBuffer = buffer;
                    oldestAge = age;
                }
            }
```